modules/phase_field/src/postprocessors/GrainTracker.C

```diff
@@ -118,12 +118,12 @@ GrainTracker::finalize()
   {
     for (const auto & grain_pair : _unique_grains)
     {
-      if (grain_pair.second->_status != INACTIVE)
+      if (grain_pair.second._status != INACTIVE)
       {
-        for (auto elem_id : grain_pair.second->_local_ids)
+        for (auto elem_id : grain_pair.second._local_ids)
         {
           mooseAssert(!_ebsd_reader || _unique_grain_to_ebsd_num.find(grain_pair.first) != _unique_grain_to_ebsd_num.end(), "Bad mapping in unique_grain_to_ebsd_num");
-          _elemental_data[elem_id].push_back(std::make_pair(_ebsd_reader ? _unique_grain_to_ebsd_num[grain_pair.first] : grain_pair.first, grain_pair.second->_var_idx));
+          _elemental_data[elem_id].push_back(std::make_pair(_ebsd_reader ? _unique_grain_to_ebsd_num[grain_pair.first] : grain_pair.first, grain_pair.second._var_idx));
```

Comment:
Ok, I see

On Fri, Jun 10, 2016 at 10:53 PM Derek Gaston <notifications@github.com>
wrote:

> In modules/phase_field/src/postprocessors/GrainTracker.C
> <https://github.com/idaholab/moose/pull/7161#discussion_r66700946>:
>
> >          {
> >            mooseAssert(!_ebsd_reader || _unique_grain_to_ebsd_num.find(grain_pair.first) != _unique_grain_to_ebsd_num.end(), "Bad mapping in unique_grain_to_ebsd_num");
> > -          _elemental_data[elem_id].push_back(std::make_pair(_ebsd_reader ? _unique_grain_to_ebsd_num[grain_pair.first] : grain_pair.first, grain_pair.second->_var_idx));
> > +          _elemental_data[elem_id].push_back(std::make_pair(_ebsd_reader ? _unique_grain_to_ebsd_num[grain_pair.first] : grain_pair.first, grain_pair.second._var_idx));
>
> emplace_back() with passing in the two arguments will be more efficient.
>
> What you have now creates two pairs, then calls the move constructor then
> destroys one of the pairs.
>
> Using emplace_back() it will simply create one pair in the vector, at the
> right spot in memory. Done.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/pull/7161/files/f76d458bebb39499d88c62360a08500cbff860ef#r66700946>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AC5XIF19zLeOTfO1CS1VrMr1DtbjIR1Qks5qKj9JgaJpZM4IwRk6>
> .
>